pd/src/wallet.rs

```diff
@@ -59,8 +60,11 @@ impl LightWallet for State {
         let (tx, rx) = mpsc::channel(100);
         tokio::spawn(
             async move {
-                for height in start_height..=end_height {
-                    let block = state.compact_block(height.into()).await;
+                while let Some(block) = state
+                    .compact_blocks(start_height.into(), end_height.into())
+                    .next()
+                    .await
+                {
```

Comment:
We have this buffered channel and a spawned task hanging around from the previous implementation, is it possible to turn the Rust `Stream` from `compact_blocks` into a gRPC stream directly?

Reply:
This doesn't work, but I think something like this could work: https://github.com/penumbra-zone/penumbra/commit/9592d5f52040c831f25e478b14aaafff4e1b592e